sql/parser/scan_test.go

```diff
@@ -267,9 +274,10 @@ func TestScanError(t *testing.T) {
 		sql string
 		err string
 	}{
-		{`1e`, "invalid floating point constant"},
-		{`1e-`, "invalid floating point constant"},
-		{`1e+`, "invalid floating point constant"},
+		{`1e`, "invalid floating point literal"},
+		{`1e-`, "invalid floating point literal"},
+		{`1e+`, "invalid floating point literal"},
+		{`0x`, "invalid hexadecimal literal"},
```

Comment:
Looking at the code, it looks like they will fail just fine, but can you add things like `0x0x...` and `00x...` just to make sure?